src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -178,6 +178,32 @@ public DynamicObject fromRope(Rope rope) {
             return executeMake(rope, NotProvided.INSTANCE, NotProvided.INSTANCE);
         }
 
+        public DynamicObject fromBuilder(RopeBuilder builder, CodeRange codeRange) {
+            return executeMake(builder.getBytes(), builder.getEncoding(), codeRange);
+        }
+
+        /**
+         * All callers of this factory method must guarantee that the builder's byte array cannot change after this
+         * call, otherwise the rope built from the builder will end up in an inconsistent state.
+         */
+        public DynamicObject fromBuilderUnsafe(RopeBuilder builder, CodeRange codeRange) {
```

Comment:
`Unsafe` seems a fairly generic term.
How about `fromBuilderNoCopy()`?

Reply:
It's the naming convention we've already using in ByteArrayBuilder (and by extension, RopeBuilder). But that aside, I think `Unsafe` conveys the risk better than `NoCopy`.

Reply:
It could be thread-unsafe, memory-unsafe, managed-unsafe, etc.
NoCopy clearly says what's unsafe about it.
But maybe this is no longer needed once we remove RopeBuilder, as it would just be a byte[] or so.